src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+                    if meta_info[1] != 'empty':
+                        logging.error("Option %s not yet implemented" % meta_info[1])
+                        img = self.parent.simpleASMCall(self._server_URL + "/scan/field", _METHOD_GET, 200, (meta_info[0].position_x,
+                                                                                                             meta_info[0].position_y, self._data_content2value(meta_info[1]).to_dict()))
+                        self.data.notify(model.DataArray(img, dtype=numpy.uint8), self._getMetadata())
+                    else:
+                        self.data.notify(model.DataArray(numpy.array([[]], dtype=numpy.uint8), self._getMetadata()))
+
+                elif command == "stop" and self._current_status != "stop":
+                    self._current_status = command
+                    self.parent.simpleASMCall(self._server_URL + "/scan/finish_mega_field", _METHOD_POST, 204)
+
+                elif command == "terminate":
+                    self._current_status = command
+                    self.parent.simpleASMCall(self._server_URL + "/scan/finish_mega_field", _METHOD_POST, 204)
+                    time.sleep(0.5)
+                    raise Exception("Terminated acquisition")
+
+                elif command == "stop" or command == "start":
+                    logging.warning("ASM acquisition was already at status '%s'" % command)
+                elif command == "next":
+                    logging.warning("Start ASM acquisition before taking field images")
+                else:
+                    logging.error("Received invalid command '%s' is skipped" % command)
+                    raise ValueError
+
+                tnow = time.time()
+                # sleep a bit to avoid refreshing too fast
+                tsleep = tnext - tnow
+                if tsleep > 0.01:
+                    time.sleep(tsleep)
+
+                tnext = time.time() + 0.2  # max 5 Hz
+
+        except Exception as e:
+            if command == "terminate":
+                raise Exception("Terminated acquisition")
+
+            elif command is not None:
+                logging.error("last message was not executed, should have perfomed action: '%s'\n"
+                              "received exepction: \n %s" % (command, e))
+
+        logging.debug("Acquisition thread ended")
+
+    def start_acquisition(self):
+        mega_field_data = self._assemble_mega_field_data()
+        self.acq_queue.put(("start", mega_field_data))
+
+    def get_next_field(self, field_num):
+        '''
+        :param field_num: x,y
+        :return:
+        '''
+        self._field_data = FieldMetaData(
+                field_num[0] * self.parent._EBeamScanner._shape.value[0],
+                field_num[1] * self.parent._EBeamScanner._shape.value[1])
+        self.acq_queue.put(("next", self._field_data, self.dataContent.value))
+
+    def stop_acquisition(self):
+        self.acq_queue.put(("stop", None))
```

Comment:
During "unloading" of the queue a tuple is expected. Since this call has no metadata, image content (empty, full or thumbnail) the added bagage is None which I have added. I should comment why I have added None.

Reply:
Have you tried this tuple for unpacking: `("stop",)`
Maybe that does the job and you don't need the None. Haven't tested it myself though.

Reply:
I thing that will give an error of not enough items to unpack, or str is not iterable. Because you try to unpack 2 values but only input 1

Reply:
I never tried it myself but saw somewhere to use a tuple with just one input. If you write it like this `("stop",)` it is a tuple and not a string. Have you tried it? Could be that it still complains about the number of values to unpack. I am curious if it would work or not :)

Reply:
Yep I did quickly tried it in the terminal, depending on wether you use int of str it will give those outputs

Reply:
The comma (I guess).
In Pyhon `("stop")` != `("stop",)`.
As Sabrina said, it's competlely fine to write `a, *b = ("stop",)` in Python 3.

Reply:
Sorry I forgot the * without that it doesn't work. But indeed sharp sabrina, this is a nice solution!